### PR TITLE
Add Entitlements / SKUs

### PR DESCRIPTION
**Pull request by jhoffi**
Implementation of [Entitlements](https://discord.com/developers/docs/monetization/entitlements#entitlement-object) and [SKUs](https://discord.com/developers/docs/monetization/skus).

### CODE REVIEW

**Comment by RazvanBerbece** [comment]
I literally just had a play with the Monetization feature on the Discord Developer Portal and was wondering how I can access SKU data and Entitlement from my app.....little did I know there's a PR for this!

Curious how this will go down - in my naivity and boredom I had a look at the changes and they seem very reasonable considering that discordgo is just a wrapper around the Discord Gateway, so I'm optimistic this will move forward!

File: restapi.go
Comment (jhoffi) [inline restapi.go:3534]: Optionally, we could pass these parameters via a struct.

File: restapi.go
Comment (jhoffi) [inline restapi.go:3535]: Optionally, we could pass these parameters via a struct.

Comment (FedorLap2006) [reply restapi.go:3535]: `appID` can stay, as it is required, everything else, yeah, into the struct.

**Comment by jhoffi** [comment]
@FedorLap2006

**Review by glotchimo — COMMENTED** [review]
Looks great, would love to see this get an owner review so it can land! Looking forward to being able to use this

File: structs.go
Comment (glotchimo) [inline structs.go:2467]: +1 for including country code type

File: structs.go
Comment (glotchimo) [inline structs.go:2512]: Is there ever a point at which we'd expect/want a default string value here? Wondering if the introduction of pointer risk is necessary or if we can roll with default values without losing any important data (no user has no ID, no guild has no ID, no entitlement has a start time at zero, etc.)

**Review by glotchimo — APPROVED** [review]
Looks good, just curious about the use of pointers in `Entitlement`

File: structs.go
Comment (jhoffi) [reply structs.go:2512]: You are right, thank you. However, I would keep all the optional timestamps as pointers, since this is also done in other structs (e.g., Channel, Member, Invite).

**Comment by ralfizzle** [comment]
Is this approved can it be merged?

**Comment by glotchimo** [comment]
> Is this approved can it be merged?

AFAIK we're waiting on maintainer approval

**Comment by glotchimo** [comment]
Bumping this again - @FedorLap2006 @bwmarrin @iopred any chance this could get a stamp? Pretty important feature set here

**Comment by iopred** [comment]
This looks great to me (but my context on discordgo is not so fresh).

I haven't merged anything in a long while, but it seems pretty reasonably reviewed by the community. Looks good to me, gl hf dd

**Comment by glotchimo** [comment]
Thanks so much!

**Pull request merged**
